testing/execution.ts

```diff
@@ -1060,6 +1099,10 @@ async function executeSyncFormulaWithContinuations({
       );
     }
     executionContext.sync.continuation = response.continuation;
+    if (response.completion) {
+      executionContext.sync.previousCompletion = response.completion as SyncCompletionMetadata<Continuation>;
```

Comment:
I'm not sure I would mutate the existing execution context. I think I'd try to return the completion, and have the caller generate a new `context.sync` value and then just call this function again with that.